hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/clustering/run/strategy/MultipleSparkJobExecutionStrategy.java

```diff
@@ -122,6 +123,8 @@ public HoodieWriteMetadata<HoodieData<WriteStatus>> performClustering(final Hood
         .stream();
     JavaRDD<WriteStatus>[] writeStatuses = convertStreamToArray(writeStatusesStream.map(HoodieJavaRDD::getJavaRDD));
     JavaRDD<WriteStatus> writeStatusRDD = engineContext.union(writeStatuses);
+    // Persist writeStatus, since it may be reused
+    writeStatusRDD.persist(StorageLevel.MEMORY_AND_DISK());
```

Comment:
we can't rely on cache since cache could be invalidated and we have had experiences where actual files written did not match w/ whats in writeStatus. Hence if something fails, we let dag to get retriggered here. we added guard rails at BasecommitActionExecutor by means of cloning. 
CC @alexeykudinkin